src/components/MealCostCalculator.vue

```diff
@@ -0,0 +1,304 @@
+        let portionCount = 0;
+
+        if ((this.meal.portions.totalWeight == "" && this.meal.portions.totalWeight == 0) || (this.meal.portions.portionWeight == "" && this.meal.portions.portionWeight == 0)) {
+          portionCount = 0;
+        } 
+        else {
+          portionCount = Math.round(totalWeight / (portionWeight / 1000));
+          this.meal.portions.portionsCount = portionCount;
+        }
+
+        portionCount < 1 ? 0 : portionCount;
+
+        return portionCount;
+      },
+
+      totalNetCost() {
+        let total = 0;
+        this.meal.ingredients.forEach(ingredient => {
+        total += parseFloat(ingredient.net);
+        });
+
+        total = total.toFixed(2);
+        this.meal.totalNetCost = total;
+      
+        return total;
+      },
+
+      totalGrossCost() {
+        let total = 0;
+        this.meal.ingredients.forEach(ingredient => {
+        total += parseFloat(ingredient.gross);
+        });
+
+        total = total.toFixed(2);
+        this.meal.totalGrossCost = total;
+
+        return total;
+      },
+
+      portionNetCost() {
+        let portion = this.meal.totalNetCost / this.meal.portions.portionsCount;
+
+        if ((this.meal.totalNetCost == "" && this.meal.totalNetCost == 0) || (this.meal.portions.portionsCount == "" && this.meal.portions.portionsCount == 0)) {
+          portion = 0;
+        } 
+        else {
+          portion = (this.meal.totalNetCost / this.meal.portions.portionsCount).toFixed(2);
+          this.meal.portions.portionNetCost = portion;
+        }
+
+        return portion;
+      },
+
+      portionGrossCost() {
+        let portion = this.meal.totalGrossCost / this.meal.portions.portionsCount;
+
+        if ((this.meal.totalGrossCost == "" && this.meal.totalGrossCost == 0) || (this.meal.portions.portionsCount == "" && this.meal.portions.portionsCount == 0)) {
+          portion = 0;
+        } 
+        else {
+          portion = (this.meal.totalGrossCost / this.meal.portions.portionsCount).toFixed(2);
+          this.meal.portions.portionGrossCost = portion;
+        }
+
+        return portion;
+      },
+  },
+
+}
+</script>
+
+<style>
+
+th, td {
+  border: none;
+  border-bottom-style: solid;
+  border-bottom-width: 1px;
+  padding-bottom: 0.5rem;
+}
+
+input{
+  border-style: solid;
+  border-width: 1px;
+  background-color: rgb(245, 245, 245);
+  padding: 0.2rem;
+}
+
+.costTable, .weightTable {
+  border: none;
+  border-bottom-style: solid;
+  border-bottom-width: 1px;
+  margin: 1rem;
+  margin-left:auto; 
+  margin-right:0;
+}
+
+.ingredientTable {
+  width: 100%;
+  border: none;
+}
+
+.tableContainer {
+  margin: 0 auto;
+  width: 80vw;
+  max-width: 1000px;
+  position: relative;
+}
+
+.dim {
+  opacity: 1;
+  transition: opacity .15s ease-in;
+}
+
+.dim:hover, .dim:focus {
+  opacity: .5;
+  transition: opacity .15s ease-in;
+}
+
+.dim:active {
+  opacity: .8;
+  transition: opacity .15s ease-out;
+}
```

Comment:
remove copy&paste code unless you want to tell me how this works

Reply:
It is not copy and paste. i will be happy to tell you